python/ray/serve/controller.py

```diff
@@ -112,13 +112,17 @@ async def __init__(
 
         self.long_poll_host = LongPollHost()
 
-        self.http_state = HTTPState(
-            controller_name,
-            detached,
-            http_config,
-            head_node_id,
-            gcs_client,
-        )
+        if os.environ.get("EXPERIMENT_DISABLE_HTTP_PROXY", "0") == "1":
+            self.http_state = None
```

Comment:
we should formalize this into some sort of configuration or merge HTTState into a regular driver deployment. i.e, HTTPProxy Deployment